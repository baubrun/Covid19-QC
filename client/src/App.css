* {
    font-family: 'Roboto', sans-serif;
}

body {
    background-color: #235789 !important;
}



