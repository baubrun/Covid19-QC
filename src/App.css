* {
    font-family: 'Roboto', sans-serif;
}

body {
    background-color: #235789;
}



.data {
    position: relative;
}

#loader {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
}

#loader {
    align-items: center;
}