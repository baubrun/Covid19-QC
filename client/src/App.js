import React from "react";
import "./App.css";
import { Title } from "./components/Title";
import { DataContainer } from "./components/DataContainer";

const App = () => {
  return (
    <div className="container">
      <Title title="Situation du Coronavirus (COVID-19) au QuÃ©bec" />
      <DataContainer />
    </div>
  );
};

export default App;
