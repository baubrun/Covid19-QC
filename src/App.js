import React, { Component } from "react";
import { RegionList } from "./components/RegionList";
import Loader from "react-loader-spinner";
import { Card } from "./components/Card";
import BarChart from "./components/BarChart/BarChart";
import "./App.css";
import { Title } from "./components/Title";
import { getData, getRegionData } from "./components/api";


class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: [],
      date: "",
      loading: false,
      region: "Total",
      currentData: [],
    };
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData = async () => {
    this.setState({
      loading: true,
    });

    try {
      const body = await getData();
      this.setState({
        data: body,
      });
      setTimeout(() => {
        this.setState({ loading: false });
        }, 900);
    } catch (error) {
      console.log(error);
    }
  };

  fetchRegionData = async (region) => {
    try {
      const body = await getRegionData(region);
      this.setState({
        currentData: body,
      });
  
    } catch (error) {
      console.log(error);
    }
  };

  regionChange = (event) => {
    const region = event.target.value;
    this.setState({
      region,
    });
    this.fetchRegionData(region);
  };


  showData = () => {
    const { data, currentData } = this.state;

    return (
      <div className=" container data">
        <div className="row">
          <div className="col">
            <Card
              data={currentData}
              title="Nombre de cas confirmés par région"
              type="confirmés"
            />
          </div>
          {
            <div className="col">
              <Card
                data={currentData}
                title="Nombre de décès par région"
                type="décès"
              />
            </div>
          }
        </div>
        <RegionList data={data} regionChange={this.regionChange} />
        <BarChart data={currentData} />
      </div>
    );
  };

  render() {
    return (
      <div className="container">
        <Title title="Situation du Coronavirus (COVID-19) au Québec" />
        {this.state.loading ? (
          <div id="loader">
            <Loader color="white" height={200} type="Puff" width={200} />
          </div>
        ) : (
          this.showData()
        )}
      </div>
    );
  }
}
export default App;
