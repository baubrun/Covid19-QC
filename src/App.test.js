import React from 'react'
import App from "./App"
import {shallow} from "enzyme"

jest.mock("./components/api")

describe('App component', () => {

    it('should Fetch data via mock', (done) => {
    });


});