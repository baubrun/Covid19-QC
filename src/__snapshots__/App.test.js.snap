// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match render snapShot 1`] = `
<div
  class="container"
>
  <div
    data-test="Title"
  >
    <h3
      class="text-center "
    >
      Situation du Coronavirus (COVID-19) au Québec
    </h3>
  </div>
  <div
    class="data"
  >
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col"
      />
    </div>
    <div
      data-test="RegionList"
    >
      <div
        class="input-group"
      >
        <select
          class="custom-select mt-3"
        >
          <option
            value="Total"
          >
            Régions
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component should match shallow snapShot 1`] = `
<div
  className="container"
>
  <Title
    title="Situation du Coronavirus (COVID-19) au Québec"
  />
  <div
    className="Spinner"
  >
    <Spinner />
  </div>
</div>
`;
